interfaces/cython/cantera/yaml2ck.py

```diff
@@ -0,0 +1,728 @@
+    if sort_elements == "alphabetical":
+        all_elements = sorted(all_elements, key=lambda e: e.symbol)
+    elif sort_elements == "atomic-mass":
+        all_elements = sorted(all_elements, key=lambda e: e.weight)
+    elif sort_elements is not None:
+        raise ValueError(
+            "sort_elements must be None, 'alphabetical', or 'molecular-weight'. "
+            f"Got '{sort_elements}'"
+        )
+
+    # Write output files
+    header_wrapper = HeaderTextWrapper(
+        [solution_name.name],
+        width=120,
+        initial_indent="! ",
+        subsequent_indent="! ",
+        break_long_words=False,
+        break_on_hyphens=False,
+    )
+    if "description" in solution.input_header:
+        header_text = header_wrapper.fill(
+            solution.input_header["description"],
+        ) + "\n"
+    else:
+        header_text = header_wrapper.fill("") + "\n"
+    mechanism_text = [
+        header_text,
+        build_elements_text(
+            all_elements,
+        ),
+        build_species_text(
+            all_species,
+        ),
+    ]
+    if thermo_path is None:
+        mechanism_text.append(
+            build_thermodynamics_text(
+                all_species,
+                separate_file=False,
+            ),
+        )
+    else:
+        thermo_text = [
+            header_text,
+            build_thermodynamics_text(
+                all_species,
+                separate_file=True,
+            ),
+        ]
+        thermo_path.write_text("\n".join(thermo_text))
+
+    # TODO: Handle phases without reactions
+    all_reactions = solution.reactions()
+    mechanism_text.append(
+        build_reactions_text(all_reactions)
+    )
+
+    if transport_path is None and transport_exists:
+        mechanism_text.append(
+            build_transport_text(all_species, separate_file=False)
+        )
+    elif transport_path is not None and transport_exists:
+        transport_text = [
+            header_text,
+            build_transport_text(all_species, separate_file=True),
+        ]
+        transport_path.write_text("\n".join(transport_text))
+
+    mechanism_path.write_text("\n".join(mechanism_text))
+
+    return output_files
+
+
+def main():
+    """
+    Parse command line arguments and pass them to `convert`
+    """
+
+    parser = argparse.ArgumentParser(
+        description="Convert Cantera YAML input files to Chemkin-format mechanisms",
+    )
+
+    parser.add_argument("input", help="The input YAML filename. Required.")
+    parser.add_argument(
+        "--phase-name", help="Name of the phase to load from the input."
+    )
+    parser.add_argument("--mechanism", help="The output mechanism filename.")
+    parser.add_argument("--thermo", help="The output thermodynamics filename.")
+    parser.add_argument("--transport", help="The output transport filename.")
+    parser.add_argument(
+        "--sort-elements",
+        choices=[None, "alphabetical", "molecular-weight"],
```

Comment:
```suggestion
        choices=[None, "alphabetical", “weight"],
```
Shorten to `weight`, as it doesn’t sound right for elements. For symmetry, also shorten for species?

Reply:
What about `molar-mass`? For both?